### ISSUE TITLE
Comments

### ISSUE THREAD

**Issue by skaae** [bordingj/Mowgly#1]
# utils.py
Ingen kommentarer

# generators.py

https://github.com/bordingj/Mowgly/blob/484ac0a9dd722657e4990b6202b682fad6e5c4ea/mowgly/training/generators.py#L58

Tror jeg ville kalde det noget i retning af: indices_sample_prob? Hvad er usecasen? Hvis man vil oversample specifikke indicer?  evt beskriv usecasen i beskrivelsen? 

https://github.com/bordingj/Mowgly/blob/484ac0a9dd722657e4990b6202b682fad6e5c4ea/mowgly/training/generators.py#L77

"train_minibatch_size must be atleast twice the size of the training data" ->
"train_minibatch_size must be **at most half** the size of the training data"

https://github.com/bordingj/Mowgly/blob/484ac0a9dd722657e4990b6202b682fad6e5c4ea/mowgly/training/generators.py#L82

Hvorfor har du det check? Teksten skal være
"the number of validation indices must be atleast **half** the number of training indices" for at passe til koden.

https://github.com/bordingj/Mowgly/blob/484ac0a9dd722657e4990b6202b682fad6e5c4ea/mowgly/training/generators.py#L83-L84

Det er en bug fordi self.val_minibatch_size er undefined. Hvorfor ganger du med 2 istedet for bare at gøre 
 ```self.val_minibatch_size = self.train_minibatch_size```

https://github.com/bordingj/Mowgly/blob/484ac0a9dd722657e4990b6202b682fad6e5c4ea/mowgly/training/generators.py#L93-L94

Er det ikke smarter at gøre 

```Python
self.pdf = pdf  # slet
self.num_threads = num_threads

if pdf is None:
    pdf = np.cumsum(np.ones(self.train_size) / float(self.train_size))

assert pdf.dtype == np.float64
assert pdf.ndim == 1
assert np.isclose(pdf.sum(), 1)
assert len(pdf) == self.val_size + self.train_size
pdf = pdf[self.train_indices]
pdf /= pdf.sum()
self.cdf_ = np.cumsum(pdf)
```
og så slette 

https://github.com/bordingj/Mowgly/blob/484ac0a9dd722657e4990b6202b682fad6e5c4ea/mowgly/training/generators.py#L108-L110

--------

https://github.com/bordingj/Mowgly/blob/484ac0a9dd722657e4990b6202b682fad6e5c4ea/mowgly/training/generators.py#L116

https://github.com/bordingj/Mowgly/blob/484ac0a9dd722657e4990b6202b682fad6e5c4ea/mowgly/training/generators.py#L124

pep8 `self.val_indices[i:i + self.val_minibatch_size]`

# meta.py

evt assert at generator_factory er klasse mowgly.training.generators.GeneratorFactory

Er det meningen at der logges loss for training og score for validation? og kan man så se om man overfitter hvis de potentielt er forskellige? Tænker bare om man ikke skal logge loss for validation også? 

https://github.com/bordingj/Mowgly/blob/484ac0a9dd722657e4990b6202b682fad6e5c4ea/mowgly/training/meta.py#L63-L65
Betyder det kør hele tiden `patience_multiplier*num_epochs` efter ny bedste score?

Hvad med learning rate decay?

**Comment by bordingj** [comment]
synes du len(pdf) == len(train_indices) eller len(pdf) == len(train_indices) + len(val_indices) og så lade constructoren indicere pdf ?

**Comment by skaae** [comment]
Jeg synes `len(pdf) == len(train_indices)`

**Comment by bordingj** [comment]
> Er det meningen at der logges loss for training og score for validation? og kan man så se om man overfitter hvis de potentielt er forskellige? Tænker bare om man ikke skal logge loss for validation også?

score_function kan jo være inverse loss_function - forhåbentlig er ens score_function inverse propotional til ens loss_function. man vi kan godt logge val_loss også hvis du synes.

> Betyder det kør hele tiden patience_multiplier*num_epochs efter ny bedste score?

ja, efter at ny score er 1/patience_increase_threshold bedre end scores_at_latest_patience_increase.

> Hvad med learning rate decay?

learning rate decay i early-stopping scheme ? hvordan det ?

**Comment by skaae** [comment]
Hvis det andet har virket for dig er det sikkert ok.

**Comment by bordingj** [comment]
har tilføjet at den reporterer val_loss
